### PR TITLE
[backport] bump helm.sh/helm/v3 from 3.10.3 to 3.11.1 (#5283)

### PR DESCRIPTION
**Pull request by jaellio**
**Description**:
Updates helm version by cherry-picking a2499eba onto release-v1.2. Also contains changes
to handle lint errors from unchecked return values (#5306).

Please answer the following questions with yes/no.

1. Does this change contain code from or inspired by another project? No
    -   Did you notify the maintainers and provide attribution?

2. Is this a breaking change? No

3. Has documentation corresponding to this change been updated in the [osm-docs](https://github.com/openservicemesh/osm-docs) repo (if applicable)? No

### CODE REVIEW

File: pkg/reconciler/client.go [hunk 1]
```diff
@@ -74,7 +74,7 @@ func (c *client) initCustomResourceDefinitionMonitor() {
 	c.informers[CrdInformerKey] = informerFactory
 
 	// Add event handler to informer
-	c.informers[CrdInformerKey].AddEventHandler(c.crdEventHandler())
+	_, _ = c.informers[CrdInformerKey].AddEventHandler(c.crdEventHandler())
```
Comment (keithmattix) [inline pkg/reconciler/client.go:77]: What are these return values?

Comment (keithmattix) [reply pkg/reconciler/client.go:77]: Oh I see it in the original PR

**Review by keithmattix — DISMISSED** [review]

File: pkg/reconciler/client.go (hunk 1 above)
Comment (steeling) [inline pkg/reconciler/client.go:77]: can we check and do something with the errors here?

Comment (jaellio) [reply pkg/reconciler/client.go:77]: Sure, I can include the updates from #5306 once that PR is merged.

Comment (jaellio) [reply pkg/reconciler/client.go]: Updated

**Review by keithmattix — APPROVED** [review]

**Pull request merged**